AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/AWSAuthSignUpOperation.swift

```diff
@@ -39,38 +36,25 @@ public class AWSAuthSignUpOperation: AmplifySignUpOperation,
             finish()
             return
         }
-        doInitialize()
-    }
-
-    func doInitialize() {
-        if isCancelled {
-            finish()
-            return
-        }
-        /// The listener is needed to cancel already ongoing sign ups and hence a getCurrentState on the state machine cannot be use to initialize. 
-        var token: AuthStateMachineToken?
-        token = stateMachine.listen { [weak self] in
-            guard let self = self else { return }
-
-            if case .configured(let authNState, _) = $0 {
+        stateMachine.getCurrentState { [weak self] in
+            guard case .configured(let authenticationState, _) = $0 else {
+                return
+            }
 
-                switch authNState {
-                case .signingUp(let authConfig, _):
-                    let event = AuthenticationEvent(eventType: .cancelSignUp(authConfig))
-                    self.stateMachine.send(event)
-                    break
-                case .signedOut:
-                    self.cancelListener(token)
-                    self.doSignUp()
-                default:
-                    self.cancelListener(token)
-                    let message = "Auth state must be signed out to signup"
-                    let error = AuthError.invalidState(message, "", nil)
-                    self.dispatch(error)
-                    self.finish()
-                }
+            switch authenticationState {
+            case .signedOut:
+                self?.doSignUp()
+            case .signingUp(let authConfig, _):
+                let event = AuthenticationEvent(eventType: .cancelSignUp(authConfig))
+                self?.stateMachine.send(event)
+                self?.doSignUp()
```

Comment:
Is there a test for this behaviour? I guess earlier implementation was not calling `self.doSignUp()`..

I think the `cancelSignUp` and `signUp` events will run concurrently, not sure if that would result in undefined behaviours.

Reply:
`self?.doSignUp()` does not start the signUp call immediately, it will wait for auth state to reach `signedOut`.